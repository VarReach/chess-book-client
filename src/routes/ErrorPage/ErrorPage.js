import React, { Component } from 'react'

export default class ErrorPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false
    };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  render() {
    if (this.state.hasError) {
      return (
        <div className="error-page">
          <h1>Something has gone wrong on our end.</h1>
          <p>Try refreshing the page in a bit.</p>
        </div>
      );
    }
    return this.props.children;
  }
}