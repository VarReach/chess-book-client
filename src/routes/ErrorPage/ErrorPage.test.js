import React from 'react';
import ReactDOM from 'react-dom';
import ErrorPage from './ErrorPage';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('ErrorPage tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ErrorPage />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  const testChildren = [<div>testing</div>];

  it('renders given children', () => {
    const wrapper = shallow(<ErrorPage children={testChildren}/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  const TestError = function() {
    return <div>Error</div>;
  }

  it('renders an error given an error', () => {
    const wrapper = shallow(
      <ErrorPage>
        <TestError />
      </ErrorPage>
    );
    wrapper.setState({ hasError: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});