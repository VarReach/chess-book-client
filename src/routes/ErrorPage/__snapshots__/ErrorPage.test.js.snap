// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage tests renders an error given an error 1`] = `
<div
  className="error-page"
>
  <h1>
    Something has gone wrong on our end.
  </h1>
  <p>
    Try refreshing the page in a bit.
  </p>
</div>
`;

exports[`ErrorPage tests renders given children 1`] = `
<div>
  testing
</div>
`;
